client/me/purchases/remove-purchase/index.jsx

```diff
@@ -0,0 +1,181 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Dialog from 'components/dialog';
+import { getPurchase, isDataLoading } from '../utils';
+import { getName, isRemovable } from 'lib/purchases';
+import Gridicon from 'components/gridicon';
+import { isDomainRegistration } from 'lib/products-values';
+import notices from 'notices';
+import purchasePaths from '../paths';
+import { removePurchase } from 'lib/upgrades/actions';
+
+const RemovePurchase = React.createClass( {
+	propTypes: {
+		selectedPurchase: React.PropTypes.object.isRequired,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
+		user: React.PropTypes.object.isRequired
+	},
+
+	getInitialState() {
+		return {
+			isDialogVisible: false,
+			isRemoving: false
+		};
+	},
+
+	closeDialog() {
+		this.setState( { isDialogVisible: false } );
+	},
+
+	openDialog( event ) {
+		event.preventDefault();
+
+		this.setState( { isDialogVisible: true } );
+	},
+
+	removePurchase( closeDialog ) {
+		this.setState( { isRemoving: true } );
+
+		const purchase = getPurchase( this.props ),
+			{ selectedSite, user } = this.props;
+
+		removePurchase( purchase.id, user.ID, success => {
+			if ( success ) {
+				if ( isDomainRegistration( purchase ) ) {
+					notices.success(
+						this.translate( `The domain {{em}}%(domain)s{{/em}} was removed from {{em}}%(wpcomUrl)s{{/em}}.`, {
+							args: {
+								domain: purchase.meta,
+								wpcomUrl: selectedSite.wpcom_url
```

Comment:
I'm not sure we need to specify that url here since the dialog for domains will show the following text:

> This will remove %(domain)s from your account. By removing, you are canceling the domain registration. This may stop you from using it again, even with another service.

I guess we should just mention _account_ again in this success message:

> The domain {{em}}%(domain)s{{/em}} was removed from your account.

Let's see what @mikeshelton1503 or @breezyskies think of that, especially since they will revisit this message for bundle domains anyway.